src/matching/matching_interface.jl

```diff
@@ -0,0 +1,158 @@
+"""
+    SSSetMatcher
```

Comment:
I would simplified the name `SSSetMatcher` to `SetMatcher`. Unless you have a good reason to add State Space, it is easier to remember.